### DISCUSSION TITLE
repositoryCache TTL

### DISCUSSION THREAD

**Discussion by wrighbr** [renovatebot/renovate#28115] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

bitbucket

### Please tell us more about your question or problem

Is there a way to extend the TTL of the repositoryCache? 
I've noticed that this cache seems to only be valid for about 15 minutes. Since we run an hourly cron job for Renovate, we're not getting the benefits of having the repositoryCache enabled.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Repository cache does not expire. Are you maybe mixing it up with the package (lookup) cache? If so then see https://docs.renovatebot.com/self-hosted-configuration/#cachettloverride

**Comment by wrighbr** [comment]
🤔 I've enabled tracing and I can see that the `extract` part is taken the longest. Will `cacheTtlOverride` help with this?

![image](https://github.com/renovatebot/renovate/assets/25573043/afe646c0-620e-4456-8498-551e3a017381)

**Reply by rarkins** [reply]
Extract normally takes up a small percentage of total time except when a repo takes a long time to clone. Renovate with repository cache enabled should be able to skip cloning if no commit has been made since the previous job, and it's almost zero time. If a commit has been made, then a clone and extract needs to be re-run.

As I mentioned before, repository cache does not expire. It's invalidated if the commit SHA changes, but otherwise lasts forever. It's up to you to make sure the repository cache files are available between jobs though.
